toxcore/group_chats.c

```diff
@@ -5118,7 +5118,7 @@ non_null(1, 2, 3, 4) nullable(7)
 static int handle_gc_custom_packet(const GC_Session *c, const GC_Chat *chat, const GC_Peer *peer, const uint8_t *data,
                                    uint16_t length, bool lossless, void *userdata)
 {
-    if (!custom_gc_packet_length_is_valid(length, lossless)) {
+    if (!(length <= (lossless ? MAX_GC_CUSTOM_LOSSLESS_INCOMING_ASSEMBLED_PACKET_SIZE : MAX_GC_CUSTOM_LOSSY_PACKET_SIZE))) {
```

Comment:
1. pull in the negation to make it `>`
2. you say "can only be less than that" but this check treats the equality as valid

Reply:
That line is too hard to read. Just break it up into a normal if/else clause.

Reply:
it was if else, then nurupo and iphy told me to fold it into this.
please make up your mind.

Reply:
Myself, nurupo and iphy are not the same person. Moreover, nurupo asked you to change it to:

`return lossless ? length <= MAX_GC_CUSTOM_LOSSLESS_INCOMING_ASSEMBLED_PACKET_SIZE
                    : length <= MAX_GC_CUSTOM_LOSSY_PACKET_SIZE;`

which is very different from:

`if (!(length <= (lossless ? MAX_GC_CUSTOM_LOSSLESS_INCOMING_ASSEMBLED_PACKET_SIZE : MAX_GC_CUSTOM_LOSSY_PACKET_SIZE))) `

Note the abundance of brackets that make it particularly difficult to parse in your head.

Reply:
now it's split up again. hope thats now more readable